### PR TITLE
ALL: Replace use of strdup with Common::String

### PR DESCRIPTION
**Pull request by ccawley2011**
>strdup is a POSIX API, not an ANSI C API. It is not available with
-std=c++11 on newlib-based systems, and VS 2015 will throw errors
unless it is #defined to alias to _strdup or unless deprecation
warnings are turned off (which is not a good idea in general).
Common::String is a safer and potentially faster (due to small
string optimisation) alternative, so prefer it instead.

Originally from PR #1128

### CODE REVIEW

**Comment by dreammaster** [comment]
Lure engine changes seem straightforward and clean

File: audio/softsynth/fluidsynth.cpp [hunk 1]
```diff
@@ -88,26 +88,15 @@ MidiDriver_FluidSynth::MidiDriver_FluidSynth(Audio::Mixer *mixer)
 }
 
 void MidiDriver_FluidSynth::setInt(const char *name, int val) {
-	char *name2 = strdup(name);
-
-	fluid_settings_setint(_settings, name2, val);
-	free(name2);
+	fluid_settings_setint(_settings, name, val);
```
Comment (sev-) [inline audio/softsynth/fluidsynth.cpp]: Nice catch. I see that in modern fluidsynth second parameter is also const.

File: graphics/fonts/bdf.cpp [hunk 2]
```diff
@@ -700,6 +700,15 @@ BdfFont *BdfFont::loadFromCache(Common::SeekableReadStream &stream) {
 	return new BdfFont(data, DisposeAfterUse::YES);
 }
 
+static char *new_strdup(const char *in) {
```
Comment (sev-) [inline graphics/fonts/bdf.cpp]: I recommend to rename it to scumm_strdup(), and move to common/str.h, similar to scumm_stricmp() and scumm_scrnicmp()

**Comment by sev-** [comment]
This is a quite good change.

Two notes, however, before it could be merged:

1. As I noted, there is a need of strdup() in one place. Thus, I recommend renaming the proposed function to scumm_strdup() and put to common/str.h as part of this patch.
2. Put strdup() to list of forbidden symbols in common/forbidden.h

**Comment by ccawley2011** [comment]
Done.

**Comment by sev-** [comment]
Great, merging.

**Pull request merged**